apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chat/item/ChatItemView.kt

```diff
@@ -1202,8 +1411,59 @@ fun moderateMessagesAlertDialog(itemIds: List<Long>, questionText: String, delet
   )
 }
 
+private fun blockAndModerateAlertDialog(
+  rhId: Long?,
+  reportedMessageId: Long,
+  reportId: Long,
+  gInfo: GroupInfo,
+  mem: GroupMember,
+  deleteMessage: suspend (Long, CIDeleteMode) -> ChatItemDeletion?
+) {
+  AlertManager.shared.showAlertDialog(
+    title = generalGetString(MR.strings.report_block_and_moderate_confirmation_title),
+    text = generalGetString(
+      if (gInfo.fullGroupPreferences.fullDelete.on) MR.strings.report_block_and_moderate_confirmation_desc_full_delete else MR.strings.report_block_and_moderate_confirmation_desc_full_delete).format(mem.chatViewName),
+    confirmText = generalGetString(MR.strings.report_block_and_moderate_confirmation_ok),
+    onConfirm = {
+      withBGApi {
+        try {
+          val deleted = deleteMessage(reportedMessageId, CIDeleteMode.cidmBroadcast)
+          if (deleted != null) {
+            blockMemberForAll(rhId, gInfo, mem, true)
+            deleteMessage(reportId, CIDeleteMode.cidmInternalMark)
+          }
+        } catch (ex: Exception) {
+          Log.e(TAG, "blockAndModerateAlertDialog block and moderate ${ex.message}")
+        }
+      }
+    },
+    destructive = true,
+  )
+}
+
 expect fun copyItemToClipboard(cItem: ChatItem, clipboard: ClipboardManager)
 
+private suspend fun getLocalIdForReportedMessage(
```

Comment:
@avently there is a similar piece of code to this in the Loader for chat items, but it deals with a lot of state that I don't have as state in this view + scrolls when done, didn't seem worth the complication of trying to wire up everything for something so small, but you may know an easy way to do it

Reply:
it's fine like this